### PR TITLE
Add `overrides` parameter on `BaseStructureConfig`, remove `merge_config`

### PR DESCRIPTION
**Pull request by vachillo**
add `overrides` to `BaseStructureConfig` to allow overriding basic fields without having to redefine them

### CODE REVIEW

File: griptape/config/azure_openai_structure_config.py [hunk 1]
```diff
@@ -96,7 +103,15 @@ class AzureOpenAiStructureConfig(StructureConfig):
...
+                LocalVectorStoreDriver, "vector_store_driver", embedding_driver=self.embedding_driver
+            ),
+            takes_self=True,
+        ),
         metadata={"serializable": True},
         kw_only=True,
     )
+
+    def __attrs_post_init__(self):
+        self.vector_store_driver = LocalVectorStoreDriver(embedding_driver=self.embedding_driver)
```
Comment (collindutter) [inline griptape/config/azure_openai_structure_config.py:117]: Can we do `Factory(lambda self: self.embedding_driver)` here?

Comment (vachillo) [reply griptape/config/azure_openai_structure_config.py:117]: there might have been an issue where the embedding driver was not getting initialized before the Factory method was being called, resulting in the vector store driver having a `DummyEmbeddingDriver`. still trying to reproduce

File: CHANGELOG.md [hunk 2]
```diff
@@ -9,9 +9,11 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 ### Added
 - `AzureOpenAiStructureConfig` for providing Structures with all Azure OpenAI Driver configuration.
 - `AzureOpenAiVisionImageQueryDriver` to support queries on images using Azure's OpenAI Vision models.
+- Parameter `overrides` to `BaseStructureConfig` to support overriding values of Structure config defaults.
 
 ### Changed
 - **BREAKING**: Updated OpenAI-based image query drivers to remove Vision from the name.
+- **BREAKING**: Removed method `merge_config` from `BaseStructureConfig`.
```
Comment (collindutter) [inline CHANGELOG.md:16]: Maybe worth mentioning "...in favor of `overrides`"

File: griptape/config/base_structure_config.py [hunk 3]
```diff
@@ -29,9 +28,17 @@ class BaseStructureConfig(BaseConfig, ABC):
...
-        base_config = self.to_dict()
-        merged_config = dict_merge(base_config, config)
+    def _factory(self, cls: type, overrides_namespace: str, **params):
+        """Factory method to create a new instance of the class with the given parameters
 
-        return BaseStructureConfig.from_dict(merged_config)
+        Args:
+            :param cls: the class to instantiate
+            :param overrides_namespace: the namespace to look for values in the overrides dictionary
+            :param params: the parameters to pass to the class
```
Comment (collindutter) [inline griptape/config/base_structure_config.py:39]: Nit the `:param ...:` syntax is not necessary

File: griptape/config/base_structure_config.py [hunk 4]
```diff
@@ -29,9 +28,17 @@ class BaseStructureConfig(BaseConfig, ABC):
...
+
+        Returns:
+            :return: a new instance of the class
```
Comment (collindutter) [inline griptape/config/base_structure_config.py:42]: Same with `:return:`. Search `Returns` for examples.

File: griptape/config/azure_openai_structure_config.py [hunk 5]
```diff
@@ -40,7 +40,9 @@ class AzureOpenAiStructureConfig(StructureConfig):
     api_key: Optional[str] = field(kw_only=True, default=None, metadata={"serializable": False})
     prompt_driver: BasePromptDriver = field(
         default=Factory(
```
Comment (collindutter) [inline griptape/config/azure_openai_structure_config.py:42]: Do we need to use attr's `Factory` now that we're doing our own? Can we somehow extend/modify the existing `Factory`?

**Review by collindutter — CHANGES_REQUESTED** [review]

**Pull request closed without merging**